metadata-integration/java/spark-lineage/src/main/java/datahub/spark/model/dataset/S3Dataset.java

```diff
@@ -0,0 +1,118 @@
+package datahub.spark.model.dataset;
+
+import java.util.List;
+import java.util.stream.Stream;
+
+import com.linkedin.common.FabricType;
+import com.typesafe.config.Config;
+
+import datahub.spark.SparkConfigUtil;
+import lombok.extern.slf4j.Slf4j;
+
+@Slf4j
+public class S3Dataset extends SparkDataset {
+
+  private static final String TABLE = "{table}";
+  private static final String TABLE_MARKER = "/{table}";
+  private static final String TABLE_MARKER_REGEX = "/\\{table\\}";
+
+  public S3Dataset(String pathUri, String platformInstance, FabricType fabricType) {
+    super("s3", platformInstance, pathUri, fabricType);
+  }
+
+  public S3Dataset(String pathUri) {
+    super("s3", pathUri);
+  }
+
+  public static boolean isMatch(String pathUri) {
+    return isS3Uri(pathUri);
+  }
+
+  public static S3Dataset create(String pathUri) throws InstantiationException {
```

Comment:
Should we throw such exception here, or maybe return Optional<S3Dataset> i.e. return value will be present if dataset could be created.
That might simplify code throughout; currently callers are having to catch this exception, and there's not really a corrective action they can take.